doc/mgr/orchestrator_modules.rst

```diff
@@ -150,6 +150,48 @@ specify a location when creating a stateless service.
 OSD services generally require a specific placement choice, as this
 will determine which storage devices are used.
 
+Error Handling
+--------------
+
+The main goal of error handling within orchestrator modules is to provide debug information to
+assist users when dealing with deployment errors.
+
+In detail, orchestrators need to explicitly deal with different kind of errors:
+
+1. No orchestrator configured
+
+.. autoclass:: NoOrchestrator
+
+2. An orchestrator doesn't implement a specific method.
+
+   For example, an Orchestrator doesn't support ``add_host``.
+
+   In this case, a ``NotImplementedError`` is raised.
+
+3. Missing features within implemented methods.
+
+   e.g. deploying iSCSI gateways is not supported
+
+.. autoclass:: MissingOrchestratorSupport
```

Comment:
For some orchestrators, this has to be encapsulatable in a `_Completion` object.

Reply:
Not sure about this example.

Presumably adding an iscsi gw would involve calling something like `ceph orchestrator iscsi add ...`

How is that different from just returning NotImplementedError ? A better example might be the case where we have optional parameters to a command that are not supported by the backend (e.g. the hosts field in update mons command with the rook backend).